File: samples/vulnerable_api.py
from fastapi import FastAPI, Depends

app = FastAPI()

# Simulação de banco de dados
class Database:
    def get(self, item_id):
        return {"item_id": item_id, "owner_id": 42, "secret_data": "Confidencial"}

db = Database()

# VULNERÁVEL: O parâmetro 'item_id' (Source) é passado diretamente para o db.get() (Sink)
# sem verificar se o usuário autenticado é o proprietário do recurso.
@app.get("/items/{item_id}")
def read_item(item_id: int):
    return db.get(item_id)

File: samples/vulnerable_api.js
const express = require('express');
const app = express();
const db = require('./db');

// VULNERÁVEL: O parâmetro ':id' (Source) é usado diretamente na query do banco (Sink)
// sem qualquer validação de propriedade ou middleware de autorização.
app.get('/api/invoice/:id', async (req, res) => {
    const invoiceId = req.params.id;
    const invoice = await db.findOne({ id: invoiceId });
    res.json(invoice);
});
